### PR TITLE
Mongoid support

### PR DESCRIPTION
**Pull request by ElvinEfendi**
Added new branch called mongoid which will allow developers to use crudify with Mongoid ODM. Als instead of will_paginate now it uses kaminari.

### CODE REVIEW

**Comment by parndt** [comment]
Why did you change the paginator as well?

**Comment by ElvinEfendi** [comment]
Because will_paginate does not support mongoid, but kaminari does. It also supports MongoMapper(someone might need in the future). As to will_paginate mongoid support, someone has already asked pull request but they have not merged it yet, you can check discussion here: https://github.com/mislav/will_paginate/pull/193

File: lib/crudify.rb [hunk 1]
```diff
@@ -1,9 +1,3 @@
-require 'active_support'
-require 'active_record'
-require 'action_controller'
-require 'will_paginate'
-require 'meta_search'
```
Comment (parndt) [inline lib/crudify.rb:5]: Why did you unrequire all of these?

File: lib/crudify/class_methods.rb [hunk 2]
```diff
@@ -41,6 +41,7 @@ def set_what
          end
                   
          def index
+	   find_all_#{plural_name}
```
Comment (parndt) [inline lib/crudify/class_methods.rb:44]: indentation wrong

File: lib/crudify/class_methods.rb [hunk 3]
```diff
@@ -131,22 +125,14 @@ def paginate_all_#{plural_name}
            # If we have already found a set then we don't need to again
            find_all_#{plural_name} if @#{plural_name}.nil?
   
-           paginate_options = {:page => params[:page]}
+	   scope = @#{plural_name}.page(params[:page])
```
Comment (parndt) [inline lib/crudify/class_methods.rb:128]: indentation wrong

File: lib/crudify/class_methods.rb [hunk 4]
```diff
@@ -113,8 +108,7 @@ def scope_with_options(scope)
   
          # Finds one single result based on the id params.
          def find_#{singular_name}
-           set_instance(#{class_name}.find(params[:id],
-             :include => #{options[:include].map(&:to_sym).inspect}))
+           set_instance(#{class_name}.includes(#{options[:include]}).find(params[:id]))
```
Comment (parndt) [inline lib/crudify/class_methods.rb:111]: :+1:

File: lib/crudify/class_methods.rb [hunk 5]
```diff
@@ -52,12 +53,6 @@ def new
   
   
          def create
-           # if the position field exists, set this object as last object, given the conditions of this class.
-           if #{class_name}.column_names.include?("position")
-             params[:#{singular_name}].merge!({
-               :position => ((#{class_name}.maximum(:position, :conditions => #{options[:conditions].inspect})||-1) + 1)
-             })
-           end
```
Comment (parndt) [inline lib/crudify/class_methods.rb:60]: Why did you remove this block of code?

File: lib/crudify/class_methods.rb [hunk 6]
```diff
@@ -131,22 +125,14 @@ def paginate_all_#{plural_name}
            # If we have already found a set then we don't need to again
            find_all_#{plural_name} if @#{plural_name}.nil?
   
-           paginate_options = {:page => params[:page]}
+	   scope = @#{plural_name}.page(params[:page])
   
-           # Seems will_paginate doesn't always use the implicit method.
-           if #{class_name}.methods.map(&:to_sym).include?(:per_page)
-             paginate_options.update(:per_page => #{class_name}.per_page)
+	   # if per page is default then set it
```
Comment (parndt) [inline lib/crudify/class_methods.rb:130]: indentation wrong

File: lib/crudify/class_methods.rb [hunk 7]
```diff
@@ -154,108 +140,11 @@ def search_all_#{plural_name}
          # from within the current controller.
          protected :find_#{singular_name},
                    :find_all_#{plural_name},
-                   :paginate_all_#{plural_name},
-                   :search_all_#{plural_name}
+                   :paginate_all_#{plural_name}
        )
   
-       # Methods that are only included when this controller is searchable.
```
Comment (parndt) [inline lib/crudify/class_methods.rb:161]: why the mass kill on searching?

**Comment by parndt** [comment]
This seems to reduce functionality for the existing ORMs.

File: lib/crudify/class_methods.rb (hunk 7 above)
Comment (ElvinEfendi) [inline lib/crudify/class_methods.rb:161]: I did not have time to port searching for mongoid

File: lib/crudify.rb (hunk 1 above)
Comment (ElvinEfendi) [inline lib/crudify.rb:5]: I did not implement searching so no need for meta_search, neither used will_paginate(instead used kaminari). Sorry for the rest seems I did mistake.
